src/pairs/GameState.py

```diff
@@ -14,14 +15,37 @@ def __init__(self, deck, target_score, players):
             self.num_players = len(self.players)
 
     def deepcopy(self):
-        c = GameState(self.deck.deepcopy(), self.target_score, self.players.deepcopy())
+        c = GameState(self.deck.deepcopy(), self.target_score,
+                      deque(player.deepcopy() for player in self.players))
         return c
 
-    def draw_next(self):
-        pass
+    def next_draw_states(self):
+        states = []
+        for card_type in self.deck.curr_deck:
+            if self.deck.curr_deck[card_type] > 0:
+                new_state = self.deepcopy()
+                new_state.deck.draw_specific_card(card_type)
+                new_state.players[0].add_card(card_type)
+                new_state.players.rotate(-1)
+                states.append(new_state)
+        return states
+
+    def next_fold_state(self):
+        curr_state = self.deepcopy()
```

Comment:
Rename `curr_state` to something like `next_state`, `fold_state`, or `next_fold_state`.

Reply:
done